src/qutip_cupy/__init__.py

```diff
@@ -36,7 +36,19 @@
         (data.Dense, CuPyDense, cd.dense_from_cupydense),
     ]
 )
-data.to.register_aliases(["cupyd"], CuPyDense)
+data.to.register_aliases(["cupyd", "CuPyDense"], CuPyDense)
+
+
+def is_cupydense(data):
+    return isinstance(data, CuPyDense)
+
+
+def is_cupy(data):
+    return isinstance(data, cp.ndarray)  # noqa: F821
```

Comment:
Is the noqa really needed now that we import cupy as cp at the top?

Reply:
The thing is that as the import is inside a try it fails to detect it

Reply:
Could you double check?

When I do:
```console
$ cat  foo.py
try:
    import math
except ImportError:
    raise


def f():
    print(math.pi)
$ flake8 foo.py
```
It doesn't print an error and in the code in this PR it doesn't print an error for `del cp` either.